fault-proof/tests/common/process.rs

```diff
@@ -0,0 +1,299 @@
+//! Process management utilities for running proposer and challenger binaries.
+
+use std::{collections::HashMap, path::PathBuf, process::Stdio, time::Duration};
+
+use anyhow::{Context, Result};
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    process::{Child, Command},
+    task::JoinHandle,
+    time::timeout,
+};
+use tracing::{error, info, warn};
+
+/// Represents a running binary process with monitoring
+pub struct ManagedProcess {
+    pub name: String,
+    pub child: Child,
+    pub stdout_handle: JoinHandle<()>,
+    pub stderr_handle: JoinHandle<()>,
+}
+
+impl ManagedProcess {
+    /// Kill the process gracefully
+    pub async fn kill(mut self) -> Result<()> {
+        info!("Stopping process: {}", self.name);
+
+        // Try graceful shutdown first
+        if let Some(pid) = self.child.id() {
+            // Send SIGTERM for graceful shutdown
+            let _ = Command::new("kill").args(["-TERM", &pid.to_string()]).output().await;
+
+            // Wait up to 5 seconds for graceful shutdown
+            match timeout(Duration::from_secs(5), self.child.wait()).await {
+                Ok(Ok(status)) => {
+                    info!("Process {} exited gracefully with status: {}", self.name, status);
+                }
+                Ok(Err(e)) => {
+                    warn!("Error waiting for process {}: {}", self.name, e);
+                }
+                Err(_) => {
+                    // Timeout - force kill
+                    info!("Process {} didn't exit gracefully, force killing", self.name);
+                    self.child.kill().await.context("Failed to kill process")?;
+                }
+            }
+        }
+
+        // Cancel output monitoring tasks
+        self.stdout_handle.abort();
+        self.stderr_handle.abort();
+
+        Ok(())
+    }
+
+    /// Check if the process is still running
+    pub fn is_running(&mut self) -> bool {
+        matches!(self.child.try_wait(), Ok(None))
+    }
+}
+
+/// Configuration for starting a binary process
+pub struct ProcessConfig {
+    pub name: String,
+    pub binary_path: PathBuf,
+    pub env_vars: HashMap<String, String>,
+    pub log_stdout: bool,
+}
+
+/// Start a proposer binary with the given configuration
+pub async fn start_proposer_binary(
```

Comment:
can we make a proposer and call proposer.run() instead? starting a binary like this will make it hard to catch errors and make it easier to gracefully stop the binary when the process exists.

Reply:
might need to make a OPSuccinctProposer::test() -> OPSuccinctProposer method or something